src/test/lit-element_test.ts

```diff
@@ -1243,26 +1310,238 @@ suite('LitElement', () => {
     container.appendChild(el);
     await el.updateComplete;
     el.foo = 'foo';
     el.id = 'id';
-    el.name = 'name';
     el.title = 'title';
     await el.updateComplete;
-    assert.equal(el.shadowRoot!.textContent, 'id-name-title-foo');
+    assert.equal(el.shadowRoot!.textContent, 'id-title-foo');
     assert.equal((window as any).id, el);
+    assert.equal(el.getAttribute('id'), 'id');
     el.id = 'id2';
-    el.name = 'name2';
     await el.updateComplete;
-    assert.equal(el.shadowRoot!.textContent, 'id2-name2-title-foo');
+    assert.equal(el.shadowRoot!.textContent, 'id2-title-foo');
     assert.equal((window as any).id2, el);
+    assert.equal(el.getAttribute('id'), 'id2');
+  });
+
+  test('user accessors correctly wrapped', async () => {
+    class Sup extends LitElement {
+      _supSet?: number;
+      _oldFoo?: any;
+      _foo?: number;
+      static get properties() { return {foo : {type : Number}}; }
+      constructor() {
+        super();
+        this.foo = 0;
+      }
+      set foo(v: number) {
+        this._supSet = (this._supSet || 0) + 1;
+        this._foo = Math.min(v, 10);
+      }
+      get foo(): number { return this._foo as number; }
+      update(changedProperties: PropertyValues) {
+        this._oldFoo = changedProperties.get('foo');
+        super.update(changedProperties);
+      }
+      render() { return html`${this.foo}`; }
+    }
+    customElements.define(generateElementName(), Sup);
+    class Sub extends Sup {
+      _subSet?: number;
+      static get properties() { return {foo : {type : Number}}; }
+      set foo(v: number) {
+        this._subSet = (this._subSet || 0) + 1;
+        super.foo = v;
+      }
+      get foo(): number {
+        const v = super.foo;
+        return v ? Math.floor(v) : v;
+      }
+    }
+    customElements.define(generateElementName(), Sub);
+
+    const sup = new Sup();
+    container.appendChild(sup);
+    await sup.updateComplete;
+    assert.equal(sup.foo, 0);
+    assert.equal(sup._oldFoo, undefined);
+    assert.equal(sup._supSet, 1);
+    assert.equal(sup.shadowRoot!.textContent, '0');
+
+    sup.foo = 5;
+    await sup.updateComplete;
+    assert.equal(sup.foo, 5);
+    assert.equal(sup._oldFoo, 0);
+    assert.equal(sup._supSet, 2);
+    assert.equal(sup.shadowRoot!.textContent, '5');
+
+    sup.foo = 20;
+    await sup.updateComplete;
+    assert.equal(sup.foo, 10); // (user getter implements a max of 10)
+    assert.equal(sup._oldFoo, 5);
+    assert.equal(sup._supSet, 3);
+    assert.equal(sup.shadowRoot!.textContent, '10');
+
+    sup.foo = 5;
+    await sup.updateComplete;
+    assert.equal(sup.foo, 5);
+    assert.equal(sup._oldFoo, 10);
+    assert.equal(sup._supSet, 4);
+    assert.equal(sup.shadowRoot!.textContent, '5');
+
+    const sub = new Sub();
+    container.appendChild(sub);
+    await sub.updateComplete;
+    assert.equal(sub.foo, 0);
+    assert.equal(sub._oldFoo, undefined);
+    assert.equal(sub._subSet, 1);
```

Comment:
Also test `sub._supSet` in this assert and the ones below.

Reply:
Done